idseq_dag/steps/run_lzw.py

```diff
@@ -1,8 +1,126 @@
+                results.append(dictionary[word])
+                dict_size += 1
+                dictionary[wc] = dict_size
+                word = c
+        if word != "":
+            results.append(dictionary[word])
+        return float(len(results)) / len(sequence)
+
+    @staticmethod
+    def generate_lzw_filtered(fasta_files, output_files, cutoff_fractions):
+        assert(len(fasta_files) == len(output_files))
+        cutoff_fractions.sort(reverse=True) # Make sure cutoff is from high to low
+
+        read_streams = []
+        readcount_list = [] # one item per cutoff
+        outstream_list = [] # one item per cutoff
+        outfiles_list = [] # one item per cutoff
+
+        for fasta_file in fasta_files:
+            read_streams.append(open(fasta_file, 'rb'))
+        paired = len(fasta_files) == 2
+
+        for cutoff in cutoff_fractions:
+            readcount_list.append(0)
+            outstream = []
+            outfiles = []
+            for f in output_files:
+                outfile_name = "%s-%f" % (f, cutoff)
+                outfiles.append(outfile_name)
+                outstream.append(open(outfile_name, 'wb'))
+
+            outstream_list.append(outstream)
+            outfiles_list.append(outfiles)
+
+        total_reads = 0
+        while True:
+            line_r1_header = read_streams[0].readline()
+            line_r1_sequence = read_streams[0].readline()
+            line_r2_header = None
+            line_r2_sequence = None
+
+            if line_r1_header and line_r1_sequence:
+                total_reads += 1
+                if paired:
+                    line_r2_header = read_streams[1].readline()
+                    line_r2_sequence = read_streams[1].readline()
+                fraction_1 = PipelineStepRunLZW.lzw_fraction(line_r1_sequence.rstrip())
+                fraction_2 = PipelineStepRunLZW.lzw_fraction(line_r2_sequence.rstrip()) if paired else 1.0
+                fraction = min(fraction_1, fraction_2)
+                for i in range(len(cutoff_fractions)):
+                    cutoff = cutoff_fractions[i]
+                    if fraction >= cutoff:
+                        readcount_list[i] += 1
+                        outstream = outstream_list[i]
+                        outstream[0].write(line_r1_header)
+                        outstream[0].write(line_r1_sequence)
+                        if paired:
+                            outstream[1].write(line_r2_header)
+                            outstream[1].write(line_r2_sequence)
+                        break
+            else:
+                break
+
+        # closing all the streams
+        for os in outstream_list:
+            for o in os:
+                o.close()
+        for r in read_streams:
+            r.close()
+        # get the right output file and metrics
+        cutoff_frac = cutoff_fractions[-1]
+        kept_count = 0
+        filtered = total_reads
+        for i in range(len(readcount_list)):
```

Comment:
You can be more pythonic: :)
`for i, readcount in enumerate(readcount_list):`

Reply:
thanks! i am a python newbie.